### PR TITLE
[Lang] Support printing scalar fields in Python-scope

### LINKED ISSUE #1126: [Lang] Matrix printing in python when shape is not ()
**Describe the bug**
Taichi seems to be unable to print the tensor of matrices.

**Log/Screenshots**
Error message 
```
  File "replicate.py", line 11, in <module>
    init()
  File "/home/peiyuanl/taichi_dev/taichi/python/taichi/lang/kernel.py", line 533, in wrapped
    return primal(*args, **kwargs)
  File "/home/peiyuanl/taichi_dev/taichi/python/taichi/lang/kernel.py", line 463, in __call__
    self.materialize(key=key, args=args, arg_features=arg_features)
  File "/home/peiyuanl/taichi_dev/taichi/python/taichi/lang/kernel.py", line 343, in materialize
    taichi_kernel = taichi_kernel.define(taichi_ast_generator)
  File "/home/peiyuanl/taichi_dev/taichi/python/taichi/lang/kernel.py", line 340, in taichi_ast_generator
    compiled()
  File "replicate.py", line 9, in init
    print(a)
  File "/home/peiyuanl/taichi_dev/taichi/python/taichi/lang/impl.py", line 327, in ti_print
    taichi_lang_core.create_print(contentries)
RuntimeError: [expr.cpp:load_if_ptr@172] ptr.cast<GlobalVariableExpression>()->snode->num_active_indices == 0
```
**To Reproduce**
```
import taichi as ti
  
a = ti.Matrix(2, 3, dt=ti.i32, shape=(1,2))
b = ti.Vector(3, dt=ti.i32, shape=(1,2))

@ti.kernel
def init():
    print(a[0,0]) # this works
    print(a) # this doesn't

init()
```

I'm wondering if this is simply a feature of Taichi....

### PR DESCRIPTION
**Pull request by archibate**
Related issue = close #1126 #1188

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
I found that we actually can print matrix fields (by @rexwangcc via `to_numpy`) but can't print scalar fields.. So adding this lost support now.

### CODE REVIEW

File: tests/python/test_print.py [hunk 1]
```diff
@@ -93,3 +93,15 @@ def func(k: ti.f32):
...
+@ti.test(arch=ti.cpu)
+def test_python_scope_print_field():
+    x = ti.Matrix.field(2, 3, dtype=ti.f32, shape=())
+    y = ti.Vector.field(3, dtype=ti.f32, shape=3)
+    z = ti.field(dtype=ti.f32, shape=3)
+
+    print(x)
+    print(y)
+    print(z)
+    ti.sync()  # TODO: move ti.sync() to @ti.test fixture
```
Comment (archibate) [inline tests/python/test_print.py:107]:
```suggestion
```

File: python/taichi/lang/expr.py [hunk 2]
```diff
@@ -197,6 +197,13 @@ def copy_from(self, other):
         assert len(self.shape) == len(other.shape)
         tensor_to_tensor(self, other)
 
+    def __repr__(self):
+        """Python scope field print support."""
+        if impl.inside_kernel():
+            return f'<Taichi Expr>'  # make pybind11 happy, see Matrix.__repr__
```
Comment (archibate) [inline python/taichi/lang/expr.py:203]:
```suggestion
            return '<Taichi Expr>'  # make pybind11 happy, see Matrix.__repr__
```

File: python/taichi/testing.py [hunk 3]
```diff
@@ -82,6 +82,15 @@ def ti_init(arch=None, exclude=None, require=None, **options):
 
 
 def test(*args, **kwargs):
+    '''
+.. function:: ti.test(arch=[], exclude=[], require=[], **options)
+
+    :parameter arch: backends to be included in test
+    :parameter exclude: backends to be excluded in test
+    :parameter require: extensions required for this test
+    :parameter options: other options to be passed into ``ti.init``
```
Comment (yuanming-hu) [inline python/taichi/testing.py:91]:
```suggestion
    :parameter arch: backends to include
    :parameter exclude: backends to exclude
    :parameter require: extensions required
    :parameter options: other options to be passed into ``ti.init``
```

**Review by Eydcao — APPROVED** [review]
Have tested both the test and printing in a normal taichi program's python's scope, LGTM!

Great work~

**Pull request merged**